packages/core/lib/events/DomainEventEmitter.ts

```diff
@@ -172,46 +162,56 @@ export class DomainEventEmitter<SupportedEvents extends CommonEventDefinition[]>
     }
 
     for (const handler of eventHandlers.foreground) {
-      const transactionId = randomUUID()
-      let isSuccessfull = false
-      try {
-        this.transactionObservabilityManager?.startWithGroup(
-          this.buildTransactionKey(event, handler, false),
-          transactionId,
-          event.type,
-        )
-        await handler.handleEvent(event)
-        isSuccessfull = true
-      } finally {
-        this.transactionObservabilityManager?.stop(transactionId, isSuccessfull)
-      }
+      await this.executeEventHandler(event, handler, false)
     }
 
-    for (const handler of eventHandlers.background) {
-      const transactionId = randomUUID()
+    const bgPromises = eventHandlers.background.map((handler) =>
```

Comment:
wonder if we should use a promise pool here to have proper concurrency and error handling. we will _probably_ have a small amount of handlers, although that is probably not a guarantee

Reply:
Not sure if promise pool will help here, we want to start all promises without waiting for them and the error handling is done per promise (if they fail we report the error and continue), maybe I am misunderstanding your message 😓